lib/omniauth/shopify/scopes.rb

```diff
@@ -0,0 +1,45 @@
+module OmniAuth
+  module Shopify
+    class Scopes
+      SCOPE_DELIMITER = ','
```

Comment:
This all seems overcomplicated when you are just wrapping a Set. We could just inherit from Set and normalize on deserialization.

```ruby
module OmniAuth
  module Shopify
    class Scopes < Set
      SCOPE_DELIMITER = ','

      def self.deserialize(scopes)
        scope_list = scopes
          .to_s
          .split(SCOPE_DELIMITER)
          .map(&:strip)
          .reject(&:empty?)
        ignore_scopes = scope_list
          .map { |scope| scope =~ /\A(unauthenticated_)?write_(.*)\z/ && "#{$1}read_#{$2}" }
          .compact
        new(scope_list - ignore_scopes)
      end

      def serialize
        to_a.join(SCOPE_DELIMITER)
      end
    end
  end
end
```

Reply:
I would be favour of this change if `normalize` is kept as a method on `Scopes` instead of building it into the deserialize method.  This will allow the caller to determine if they want a normalized value even if they don't have original serialized string.

Reply:
I agree. I think it's best to leave `normalize` that is up to the caller on whether they need it. I just made the refactor on my end and that approach works.